### ISSUE TITLE
`InputTime` dropdown navigation improvements

### ISSUE THREAD

**Issue by pixelbandito** [cision/rover-ui#251]
Relates to #203

- dropdown, when shown, should be keyboard navigable. Should take up and down actions, string that hops to the first matching option. If on click, escape makes it disappear, if on focus, escape isn't handled by the input.
- typeahead-like should scroll the the dropdown select the closest match based on partially user-entered text.
